app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -525,19 +525,19 @@ public function generateHtml(Form $entity, $persist = true)
         $html               = $this->templatingHelper->getTemplating()->render(
             $theme.'MauticFormBundle:Builder:form.html.php',
             [
-                'fieldSettings'  => $this->getCustomComponents()['fields'],
-                'viewOnlyFields' => $this->getCustomComponents()['viewOnlyFields'],
-                'fields'         => $fields,
-                'contactFields'  => $this->leadFieldModel->getFieldListWithProperties(),
-                'companyFields'  => $this->leadFieldModel->getFieldListWithProperties('company'),
-                'form'           => $entity,
-                'theme'          => $theme,
-                'submissions'    => $submissions,
-                'lead'           => $lead,
-                'formPages'      => $pages,
-                'lastFormPage'   => $lastPage,
-                'style'          => $style,
-                'inBuilder'      => false,
+                'fieldSettings'          => $this->getCustomComponents()['fields'],
+                'doNotProgressiveFields' => $this->getDoNotProgressiveFields(),
```

Comment:
The current way this is implemented introduces a breaking change, unfortunately. The old version, `$this->getCustomComponents()['viewOnlyFields']`, has functionality where it gets the form fields from the event. Fields can have `['builderOptions']['addSaveResult']` set to `false`, which will add them to the `$notPersist` array and therefore not show them:

https://github.com/mautic/mautic/blob/ead123f149755d6c9c42425da22c9f9e97b5daef/app/bundles/FormBundle/Model/FormModel.php#L716-L731

... this PR breaks that functionality because here we work with a hardcoded list from `getDoNotProgressiveFields()`. 

You should be able to do something like

```suggestion
                'doNotProgressiveFields' => [...$this->getCustomComponents()['viewOnlyFields'], 'hidden'],
```

... which won't cause a BC break and still exclude the hidden fields from the view 😊 The alternative would be to introduce `doNotProgressiveFields` as an addition next to `viewOnlyFields` which is also no BC, but I'm not sure if that would make a lot of sense. Have just looked at this from a code perspective.

Reply:
The reason to create `doNotProgressiveFields` list was addSaveResult. If you set it to false, then as I said are added to viewOnlyFields. But in same cases customers do want use it and use it also in progressive proffiling. It make conflicts for us and BC, some forms stop working..